store.go

```diff
@@ -3103,13 +3109,21 @@ func (s *store) CleanupStagedLayer(diffOutput *drivers.DriverWithDifferOutput) e
 	return err
 }
 
+func (s *store) PrepareStagedLayer(options *drivers.ApplyDiffWithDifferOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error) {
+	return s.ApplyDiffWithDiffer("", options, differ)
```

Comment:
(Non-blocking: Consider switching the direction of the indirection, so that the deprecated `ApplyDiffWithDiffer` calls the new `PrepareStagedLayer`. That way, there won’t be any callers of ADWD remaining, and removing it later will be a smaller change.

LGTM either way.)

Reply:
the check for `rlstore.Exists(to)` requires to take a lock.

If you prefer I could duplicate the code:

```go
func (s *store) PrepareStagedLayer(options *drivers.ApplyDiffWithDifferOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error) {
	return writeToLayerStore(s, func(rlstore rwLayerStore) (*drivers.DriverWithDifferOutput, error) {
		return rlstore.ApplyDiffWithDiffer(options, differ)
	})
}
```

Reply:
yeah sorry, I was looking at doing this change in earlier commits from the PR, but I can apply it directly to the "chunked: fix reuse of the layers cache" patch.

Applied now, and pushed a new version